acceptance/support/lsrs-rest-client.ts

```diff
@@ -0,0 +1,165 @@
+     */
+    public async callDownloadFileEndpoint(): Promise<AxiosResponse<FileModel>> {
+
+        const pathUUID: string = DataRegistry.get(Attribute.PATH_UUID);
+        expect(pathUUID.includes("/")).toBe(true);
+
+        return this.axiosGet(`files/${pathUUID}`);
+    }
+
+    /**
+     * Calls the PUT /files/:pathUUID endpoint and returns its response.
+     */
+    public async callUpdateFileMetaInfoEndpoint(): Promise<AxiosResponse<FileModel>> {
+
+        const pathUUID = DataRegistry.get(Attribute.PATH_UUID);
+        const updateRequest: UploadedFileUpdateAttributes = {
+            originalFilename: DataRegistry.get(Attribute.FIELD_ORIGINAL_FILENAME),
+            description: DataRegistry.get(Attribute.FIELD_DESCRIPTION)
+        };
+
+        return await axios.put(`${this.baseURL}/files/${pathUUID}`, updateRequest, this.getAxiosRequestConfig());
+    }
+
+    /**
+     * Calls the POST /files endpoint and returns its response.
+     */
+    public async callUploadFileEndpoint(): Promise<AxiosResponse<FileModel>> {
+
+        const uploadRequest = new FormData();
+        if (DataRegistry.get(Attribute.FIELD_INPUT_FILE)) {
+            uploadRequest.append(Attribute.FIELD_INPUT_FILE, DataRegistry.get(Attribute.FIELD_INPUT_FILE), DataRegistry.get(Attribute.FIELD_ORIGINAL_FILENAME));
+        }
+        uploadRequest.append(Attribute.FIELD_SUBFOLDER, DataRegistry.get(Attribute.FIELD_SUBFOLDER) ?? "");
+        uploadRequest.append(Attribute.FIELD_DESCRIPTION, DataRegistry.get(Attribute.FIELD_DESCRIPTION) ?? "");
+
+        const requestConfig = this.getAxiosRequestConfig(uploadRequest.getHeaders());
+        requestConfig.transformRequest = axios.defaults.transformRequest;
+        requestConfig.transformResponse = axios.defaults.transformResponse;
+
+        return await axios.postForm(`${this.baseURL}/files`, uploadRequest, requestConfig);
+    }
+
+    /**
+     * Calls the POST /files/directories endpoint and returns its response.
+     */
+    public async callCreateDirectoryEndpoint(): Promise<AxiosResponse<FileModel>> {
+
+        const createDirectoryRequest: { [key in keyof DirectoryCreationRequestModel]: string } = {
+            parent: DataRegistry.get(Attribute.FIELD_PARENT),
+            name: DataRegistry.get(Attribute.FIELD_NAME)
+        };
+
+        return await axios.post(`${this.baseURL}/files/directories`, createDirectoryRequest, this.getAxiosRequestConfig());
+    }
+
+    /**
+     * Calls the DELETE /files/:pathUUID endpoint and returns its response.
+     */
+    public async callDeleteFileEndpoint(): Promise<AxiosResponse<FileModel>> {
+
+        const pathUUID = DataRegistry.get(Attribute.PATH_UUID);
+
+        return await axios.delete(`${this.baseURL}/files/${pathUUID}`, this.getAxiosRequestConfig());
+    }
+
+    /**
+     * Returns the initialized LSRSRestClient instance (initializes first if needed).
+     */
+    public static async getInstance(): Promise<LSRSRestClient> {
```

Comment:
Why is this async?

Reply:
Dynamic import leftover.. nice catch, thanks, will fix!